Feat/support media queries (#9)
vtex-apps/responsive-values: docs/README.md

# vtex.responsive-values

<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->

[![All Contributors](https://img.shields.io/badge/all_contributors-0-orange.svg?style=flat-square)](#contributors-)

<!-- ALL-CONTRIBUTORS-BADGE:END -->

> Utility for using props that accept different values for different devices or media queries.

Reviewer: ```suggestion
In VTEX IO, some component's props, such as the ones from the [Flex-Layout](https://developers.vtex.com/docs/vtex-flex-layout) app, support responsive-values. That means it's possible to assign, to the same prop, different values for each device type or specific media queries.

Props that support responsive values accept either a unique value for all cases or an object containing specific definitions for each of the following: `phone`, `tablet`, `desktop`, `mobile` (for both phone and tablet), and media queries.
```

}
```

> ...or different values depending on the device or a media query:

Reviewer: ```suggestion
... or different values depending on the device type or specific media queries:
```

/> {/* returns 1 on mobile devices, 2 on desktop */}
```

> There are some things to keep in mind when using media queries:
> 
> - Media queries take precedence over devices. If one matches, any device-specific value is ignored.
> - Media queries are written following the [standard media query syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Syntax).
> - The definiton order matters. If more than one media query is matched, the first matched one is used.

Reviewer: ```suggestion
Considering the use of a device, which responsive values are missing, the page style will match the following largest screen size definition.

For example, if a component has only `phone` and `desktop` responsive values defined, tablet devices will receive the value from `desktop`. 

Media queries, in turn, have no fallback behavior. Hence, when using this feature, it's important to keep in mind that:

- Media queries take precedence over devices. Therefore, if a media query matches the current state of a window, all device definitions are ignored.
- The order definition matters. Therefore, if more than one media query matches the current state of a window, the first match is used.
- Media queries are written following the [standard media query syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Syntax).
```

> ## API
> 

Reviewer: ```suggestion
```

> Props that use this hook accept either a value by itself, or an object with the following options: `phone` and `tablet` separately (or just `mobile` for both), `desktop` and generic media queries.
> 
> If there is any missing device, it will fallback to the next largest one. For example, if only the values of `phone` and `desktop` are passed, `tablet` devices will receive the value from `desktop`. Media queries have no fallback behavior.
> 

Reviewer: ```suggestion
```
